schemas/com.mailgun/message_clicked/jsonschema/1-0-0

```diff
@@ -0,0 +1,122 @@
+      "type": "string",
+      "maxLength": 64
+    },
+    "client-os": {
+      "description": "OS family running the client software, e.g. ‘Linux’, ‘Windows’, ‘OSX’.",
+      "type": "string",
+      "maxLength": 64
+    },
+    "client-type": {
+      "description": "Type of software the link was opened in, e.g. ‘browser’, ‘mobile browser’, ‘email client’.",
+      "type": "string",
+      "maxLength": 64
+    },
+    "country": {
+      "description": "Two-letter country code (as specified by ISO3166) the event came from or ‘unknown’ if it couldn’t be determined.",
+      "type": "string",
+      "maxLength": 2
+    },
+    "device-type": {
+      "description": "Device type the link was clicked on. Can be ‘desktop’, ‘mobile’, ‘tablet’, ‘other’ or ‘unknown’.",
+      "type": "string",
+      "maxLength": 64
+    },
+    "domain": {
+      "description": "Domain that sent the original message.",
+      "type": "string",
+      "maxLength": 256
+    },
+    "event": {
+      "description": "Event name ('clicked').",
+      "type": "string",
+      "maxLength": 16
+    },
+    "ip": {
+      "description": "IP address the event originated from (currently only seems to support ipv4).",
+      "type": "string",
+      "format": "ipv4"
+    },
+    "mailing-list": {
+      "description": "The address of mailing list the original message was sent to.",
+      "type": "string"
+    },
+    "message-id": {
+      "description": "The Id of the message that bounced.",
+      "type": "string",
+      "maxLength": 1024
+    },
+    "recipient": {
+      "description": "Recipient who clicked.",
+      "type": "string",
+      "maxLength": 512
+    },
+    "region": {
+      "description": "Two-letter or two-digit region code or ‘unknown’ if it couldn’t be determined.",
+      "type": "string",
+      "maxLength": 8
+    },
+    "signature": {
+      "description": "String with hexadecimal digits generate by HMAC algorithm (see securing webhooks).",
+      "type": "string",
+      "pattern": "^([A-Za-z0-9+/]{4})*([A-Za-z0-9+/]{4}|[A-Za-z0-9+/]{3}=|[A-Za-z0-9+/]{2}==)$",
+      "maxLength": 128
+    },
+    "tag": {
+      "description": "Message tag, if it was tagged. See Tagging.",
+      "type": "string"
+    },
+    "timestamp": {
+      "description": "Number of seconds passed since January 1, 1970 (see securing webhooks).",
+      "type": "integer",
+      "maximum": 2147483647
+    },
+    "token": {
+      "description": "Randomly generated string with length 50 (see securing webhooks).",
+      "type": "string",
+      "maxLength": 64
+    },
+    "url": {
+      "description": "The URL that was clicked (size from rfc7230#section-3.1.1).",
+      "type": "string",
+      "format": "uri",
+      "maxLength": 8192
+    },
+    "user-agent": {
+      "description": "User agent string of the client triggered the event.",
+      "type": "string", 
+      "maxLength": 1024
+    }
+  },
+  "additionalProperties": true
```

Comment:
Nothing `required`?

Reply:
It is not clear to me what is required from the documentation.